airflow/models/renderedtifields.py

```diff
@@ -49,6 +50,7 @@ def __init__(self, ti: TaskInstance, render_templates=True):
         self.ti = ti
         if render_templates:
             ti.render_templates()
+        self.k8s_pod_yaml = ti.render_k8s_pod_yaml()
```

Comment:
This is going to cause an exception when the Kube modules aren't available.

```suggestion
        if is_k8s_or_k8scelery_executor():
            self.k8s_pod_yaml = ti.render_k8s_pod_yaml()
```